노욱진/index.js

```diff
@@ -0,0 +1,36 @@
+const buttons = document.querySelectorAll("button");
+const displayFormula = document.querySelector(".js-cal-display-formula");
+const display = document.querySelector(".js-cal-display");
+
+const resultButton = document.querySelector(".js-result");
+const resetButton = document.querySelector(".js-reset");
+
+const INIT_NUM = "0";
+
+const resultNumber = () => {
+  return (display.innerHTML = eval(display.innerHTML));
+};
+
+const resetNumber = () => {
+  return (display.innerHTML = INIT_NUM), (displayFormula.innerHTML = INIT_NUM);
+};
+
+const init = () => {
+  buttons.forEach((value) => {
+    value.addEventListener("click", (event) => {
+      const { target } = event;
+      const { value } = target;
+
+      if (displayFormula.innerHTML && display.innerHTML === INIT_NUM) {
+        (displayFormula.innerHTML = value), (display.innerHTML = value);
+      } else {
+        (displayFormula.innerHTML += value), (display.innerHTML += value);
+      }
+
+      resultButton.addEventListener("click", resultNumber);
+      resetButton.addEventListener("click", resetNumber);
+    });
+  });
+};
```

Comment:
```suggestion
const init = () => {
  resultButton.addEventListener("click", resultNumber);
  resetButton.addEventListener("click", resetNumber);
  buttons.forEach((value) => {
    value.addEventListener("click", (event) => {
      const { target } = event;
      const { value } = target;

      if (displayFormula.innerHTML && display.innerHTML === INIT_NUM) {
        (displayFormula.innerHTML = value), (display.innerHTML = value);
      } else {
        (displayFormula.innerHTML += value), (display.innerHTML += value);
      }
    });
  });
};
```

중괄호가 헷갈리셔서 실수하신 거 같아요 욱진님 ㅋㅋㅋㅋ

지금 작성해 주신 코드라면 버튼들이 클릭할 때마다 이벤트 리스너가 부착되는 모습이 됩니다 !

> 단적으로 버튼을 누르기 전에 result 버튼을 누른다면 아무 동작을 하지 않게 되겠죠 

중괄호가 헷갈리시다면 vsc 기능 중에 중괄호 짝끼리 색을 맞춰주는 기능이 있는데 링크 남겨드릴게요

https://violetboralee.medium.com/vscode%EC%97%90%EC%84%9C-bracket-pair-colorizer-%EC%82%AD%EC%A0%9C%ED%95%98%EA%B3%A0-%EB%82%B4%EC%9E%A5-%EA%B8%B0%EB%8A%A5-%EC%82%AC%EC%9A%A9%ED%95%98%EA%B8%B0-e75558a87990

Reply:
바로 설치했습니다..ㅋㅋㅋㅋㅋㅋ 👍 👍